### PR TITLE
Fix the data type

### PR DESCRIPTION
**Pull request by SnowNooDLe**
# Fix the data type

## Updated on 24th November,
Those issues were found while plotting verification plots for IM vs Rrup which only has a single period value for pSA.

### `median = median[0] if isinstance(median, Iterable) else median` usage
For BCH_16 model, 
```python
sa_int = np.interp(np.log(period), np.log(periods_int), as)
```

This returns `ndarray` for this model [np.interp](https://numpy.org/doc/stable/reference/generated/numpy.interp.html), based on the doc, there are three different types of return? but its `ndarray` for our use case.
Screenshot from 2021-11-24 16-29-48
### This is what other models return 
Screenshot from 2021-11-24 16-30-05

### `if len(res) == 1:  median, (sigma, _, _) = res[0]` usage
With BR_10 model, even if there is only one period for pSA, it still puts inside a list.
Screenshot from 2021-11-24 16-31-29

## Previous description
During the verification plots process, noticed that the `medians` is a list of mixed data types.

Following parameters were used when this error was triggered.
`pSA` with `SUBDUCTION_SLAB`, 

image (8)

With `np.log()` for `logmedians`, we get the following error message
```python
TypeError: loop of ufunc does not support argument 0 of type int which has no callable log method
```

For `logsigmas`, getting the following error message
```python
VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  return array(a, dtype, copy=False, order=order)
```

Adding the following lines to make sure to append float to `medians` instead of any other data type (E.g., list, tuple or np.ndarray)
```python
if isinstance(median, Iterable):
    median = median[0]
if isinstance(sigma, Iterable):
    sigma = sigma[0]
```

### CODE REVIEW

**Review by jasonmotha — DISMISSED** [review]

**Comment by SnowNooDLe** [comment]
@jasonmotha @jpa198 
Just a minor thought, as I suggested to have this and something came up this morning,
Would it be useful/futureproof to have the following code instead?
```python
if isinstance(median, Iterable):
    for item in median:
        medians.append(item) 
else:
    medians.append(median)

if isinstance(sigma, Iterable):
    for item in sigma:
        sigmas.append(item)
else:
    sigmas.append(sigma)
```
Because the following code, if `res` is not a tuple, both `median` and `sigma` are going to be in a list. However, based on what @jpa198 told me, that when `res` is not a tuple, there is a chance that `res` has more than one iterable item in it.
(I guess something with pSA, if you pass a list of periods, it may return multiple medians/sigmas along with a different period value?)
And with the suggestion I made, it would work if there is only one median/sigma in a list but if there are more than one, will not work.
```python
if isinstance(res, tuple):
    median, (sigma, _, _) = res
else:
    median = [x[0] for x in res]
    sigma = [x[1][0] for x in res]
```

**Review by jpa198 — APPROVED** [review]

**Comment by jasonmotha** [comment]
@SnowNooDLe Good point. For pSA it will be a list of results. Though for this change I'd recommend `extend` rather than appending by element.

File: empirical/GMM_models/meta_model.py [hunk 1]
```diff
@@ -53,9 +55,14 @@ def meta_model(fault, site, im, weights_path=None, period=None, config=None, **k
         )
         if isinstance(res, tuple):
             median, (sigma, _, _) = res
+            median = median[0] if isinstance(median, Iterable) else median
```
Comment (jpa198) [inline empirical/GMM_models/meta_model.py:58]: Does sigma need a line like this?

**Review by jpa198 — APPROVED** [review]

File: empirical/GMM_models/meta_model.py (hunk 1 above)
Comment (SnowNooDLe) [reply empirical/GMM_models/meta_model.py:58]: @jpa198 Yeah, when Meta contains more models, might need another investigation but for now
When tect_type is SUBDUCTION_SLAB and IM is pSA, with a single period,
BCH_16 models returns in the form of
```python
# Array and tuple inside a tuple
([], ())
```
Every other model return in the form of
```python
# Float and tuple inside a tuple
(0.xxx, ())
```
image

Comment (SnowNooDLe) [reply empirical/GMM_models/meta_model.py:58]: Ah, replied against median not sigma,
With the current models inside the Meta config, no, we do not need them, they always return in a tuple form.

**Review by jasonmotha — APPROVED** [review]

**Review by jpa198 — APPROVED** [review]

**Pull request merged**